apps/prairielearn/assets/scripts/lib/tabulator.ts

```diff
@@ -0,0 +1,125 @@
+      `Showing ${currentRow}-${currentRow + pageSize - 1} of ${totalRows} question${
+        totalRows === 1 ? '' : 's'
+      }` +
+      (totalRows === table.getData().length
+        ? ''
+        : ` (filtered from ${table.getData().length} total question${
+            table.getData().length === 1 ? '' : 's'
+          })`),
+    paginationSize: 50,
+    paginationSizeSelector: [10, 20, 50, 100, 200, 500, true],
+    ...options,
+  });
+
+  if (options.columnVisibilityDropdown) {
+    const dropdown = options.columnVisibilityDropdown;
+    table.on('tableBuilt', () => {
+      table.getColumns().forEach((col) => {
+        const dropdownItem = parseHTMLElement(
+          document,
+          html`<div class="dropdown-item form-check">
+            <label class="form-check-label">
+              <input class="form-check-input" type="checkbox" />${col.getDefinition().title}
+            </label>
+          </div>`,
+        );
+        dropdown.appendChild(dropdownItem);
+        const input = dropdownItem.querySelector<HTMLInputElement>('input');
+        if (input != null) {
+          input.checked = col.isVisible();
+          input.addEventListener('change', () => {
+            input.checked ? col.show() : col.hide();
+            table.redraw();
+          });
+        }
+      });
+    });
+  }
+
+  if (options.layout == null) {
+    // Only rearrange column widths if the layout is not explicitly set to a different value
+    table.on('renderStarted', () => {
+      // Reset column widths to fit content
+      table
+        .getColumns()
+        .filter((col) => col.getWidth() != null)
+        .forEach((col) => {
+          col.setWidth(true);
+        });
+    });
+
+    table.on('renderComplete', () => {
+      // Resize columns to fill the table width (since fitData does not support this)
+      const columnsWidth = table
+        .getColumns()
+        .map((col) => col.getWidth())
+        .reduce((a, b) => a + (b ?? 0), 0);
+      const tableWidth = table.element.clientWidth;
+      const extraWidth = tableWidth - columnsWidth;
+      if (extraWidth > 0) {
+        const ratio = 1 + extraWidth / columnsWidth;
+        table
+          .getColumns()
+          .filter((col) => col.getWidth() != null)
+          .forEach((col) => {
+            col.setWidth(col.getWidth() * ratio);
+          });
+      }
+    });
+  }
+
+  return table;
+}
+
+export function selectFilterOnSearch(table: Tabulator): void {
+  document.addEventListener('keydown', (event) => {
+    if ((event.ctrlKey || event.metaKey) && event.key === 'f') {
+      // Set focus to filter of first visible column with a filter
+      const firstVisibleColumn = table
+        .getColumns()
+        .find((c) => c.isVisible() && c.getDefinition().headerFilter != null);
+      if (firstVisibleColumn != null) {
+        table.setHeaderFilterFocus(firstVisibleColumn);
+        event.preventDefault();
```

Comment:
I sort of hate it when pages hijack `Ctrl-F` 😬 can we find a middle ground and make it so that pressing Ctrl-F twice in a row still triggers the native browser one? I think you can do this by allowing the default to happen if the filter element is already focused (`=== document.activeElement`).

Reply:
Done in ca65b122eeff1a1310244622eb77b503f2167427. Can you check if the behaviour matches what you have in mind?